calendarsubscription/store.go

```diff
@@ -0,0 +1,152 @@
+package calendarsubscription
+
+import (
+	"context"
+	"database/sql"
+	uuid "github.com/satori/go.uuid"
+	"github.com/target/goalert/permission"
+	"github.com/target/goalert/util"
+	"github.com/target/goalert/util/sqlutil"
+	"github.com/target/goalert/validation/validate"
+	"time"
+)
+
+type Store interface {
+	FindOne(context.Context, string) (*CalendarSubscription, error)
+	CreateSubscriptionTx(context.Context, *sql.Tx, *CalendarSubscription) (bool, error)
+	UpdateSubscriptionTx(context.Context, *sql.Tx, *CalendarSubscription) error
+	DeleteSubscriptionsTx(context.Context, *sql.Tx, []string) error
+	Search(context.Context, *SearchOptions) ([]CalendarSubscription, error)
+}
+
+type DB struct {
+	db *sql.DB
+
+	findOne *sql.Stmt
+	create  *sql.Stmt
+	update  *sql.Stmt
+	delete  *sql.Stmt
+}
+
+func NewDB(ctx context.Context, db *sql.DB) (*DB, error) {
```

Comment:
nit: Let's try to add comments for public/exported methods as part of the dev life work